pkg/sitewise/auth_test.go

```diff
@@ -74,3 +161,51 @@ func TestAuthWithServer(t *testing.T) {
 	require.Equal(t, settings.EdgeAuthMode, info.AuthMechanism)
 	require.Equal(t, settings.EdgeAuthUser, info.Username)
 }
+
+type AuthRequest struct {
+	Username      string `json:"username,omitempty"`
+	Password      string `json:"password,omitempty"`
+	AuthMechanism string `json:"authMechanism,omitempty"`
+}
+
+// helper function to create a certificate template with a serial number and other required fields
+func CertTemplate() (*x509.Certificate, error) {
+	// generate a random serial number (a real cert authority would have some logic behind this)
+	serialNumberLimit := new(big.Int).Lsh(big.NewInt(1), 128)
+	serialNumber, err := rand.Int(rand.Reader, serialNumberLimit)
+	if err != nil {
+		return nil, errors.New("failed to generate serial number: " + err.Error())
+	}
+
+	tmpl := x509.Certificate{
+		SerialNumber:          serialNumber,
+		Subject:               pkix.Name{Organization: []string{"Test Inc."}},
+		SignatureAlgorithm:    x509.SHA256WithRSA,
+		NotBefore:             time.Now(),
+		NotAfter:              time.Now().Add(time.Hour), // valid for an hour
+		BasicConstraintsValid: true,
+	}
+	return &tmpl, nil
+}
+
+//helper function to create a certificate from a template and public key plus a parent certificate and private key
+func CreateCert(template, parent *x509.Certificate, pub interface{}, parentPriv interface{}) (
```

Comment:
```suggestion
func createCert(template, parent *x509.Certificate, pub interface{}, parentPriv interface{}) (
```

keep private unless required